build.py

```diff
@@ -214,13 +295,41 @@ def _validate_android_args(args: argparse.Namespace):
                 )
 
         if args.cmake_generator != original_cmake_generator:
-            log.info(f"Setting CMake generator to '{args.cmake_generator}' for cross-compiling for Android.")
+            log.info(
+                f"Setting CMake generator to '{args.cmake_generator}' for cross-compiling for Android."
+            )
 
         # no C# on Android so automatically skip
         if not args.skip_csharp:
             args.skip_csharp = True
 
 
+def _validate_ios_args(args: argparse.Namespace):
+    if args.ios:
+        # TODO: cross compile for iOS on Linux?
+        if util.is_mac():
+            needed_args = [
+                args.ios_sysroot,
+                args.osx_arch,
+                args.apple_deployment_target,
+            ]
+            arg_names = [
+                "--ios_sysroot          "
+                + "<the location or name of the macOS platform SDK>",
+                "--osx_arch             "
+                + "<the Target specific architectures for iOS>",
+                "--apple_deploy_target  "
+                + "<the minimum version of the target platform>",
+            ]
+            if not all(_ is not None for _ in needed_args):
```

Comment:
nit: would avoiding a double negative be a little easier to read?

```suggestion
            have_required_args = all(_ is not None for _ in needed_args)
            if not have_required_args:
```